src/blockchain/types/address.rs

```diff
@@ -0,0 +1,191 @@
+    /// Creates an address with the address given as 20 bytes.
+    pub fn from_bytes(bytes: [u8; 20]) -> Self {
+        Address(bytes)
+    }
+
+    /// Creates an address from a string in hex format of the address.
+    ///
+    /// *Arguments*
+    ///
+    /// * `string` - A Strting in hex format that represents 20 bytes.
+    ///              Must be exactly 40 characters long. Any leading `0x` will be removed.
+    pub fn from_string(string: &str) -> Result<Self, Error> {
+        let mut cleaned = &string.to_string()[..];
+        cleaned = cleaned.trim();
+
+        // cut leading "0x" if present
+        if &cleaned[..2] == "0x" {
+            cleaned = &cleaned[2..];
+        }
+
+        if cleaned.len() != 40 {
+            return Err(Error::new(
+                ErrorKind::InvalidAddress,
+                format!(
+                    "Expected 40 characters. Got {} instead: {}",
+                    cleaned.len(),
+                    cleaned
+                ),
+            ));
+        }
+
+        // Convert byte by byte.
+        let mut bytes = [0u8; 20];
+        let mut index = 0;
+        loop {
+            let byte = u8::from_str_radix(&cleaned[..2], 16);
+            let byte = match byte {
+                Ok(byte) => byte,
+                Err(error) => {
+                    return Err(Error::new(
+                        ErrorKind::InvalidAddress,
+                        format!("Could not parse hex string into address bytes: {}", error),
+                    ))
+                }
+            };
+
+            bytes[index] = byte;
+
+            cleaned = &cleaned[2..];
+            if cleaned.len() < 2 {
+                break;
+            }
+            index += 1;
+        }
+
+        Ok(Address(bytes))
+    }
+
+    /// Returns the bytes representation of this address.
+    pub fn bytes(&self) -> [u8; 20] {
+        self.0
+    }
+}
+
+impl PartialEq for Address {
+    /// Two addresss are equal if their byte representations are equal.
+    fn eq(&self, other: &Address) -> bool {
+        self.bytes() == other.bytes()
+    }
+}
+
+impl LowerHex for Address {
+    /// Writes the bytes as hex with leading zeros to the given Formatter.
+    fn fmt(&self, f: &mut Formatter) -> Result<(), fmt::Error> {
+        for byte in &self.0 {
+            write!(f, "{:02x}", byte)?;
+        }
+
+        Ok(())
+    }
+}
+
+impl Debug for Address {
+    /// Debug format is lower hex.
+    fn fmt(&self, f: &mut Formatter) -> Result<(), fmt::Error> {
+        write!(f, "{:x}", self)?;
+
+        Ok(())
+    }
+}
+
+// Converting other types to and from Addresss
+pub trait AsAddress {
+    fn as_address(&self) -> Address;
+}
+
+pub trait FromAddress {
+    fn from_address(address: Address) -> Self;
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn address_from_string() {
+        let mut address =
+            Address::from_string(&"0000000000000000000000000000000000000000".to_owned()).unwrap();
```

Comment:
:)  -- I think `Address::from_string("0000000000000000000000000000000000000000").unwrap();` does the trick 💃

Reply:
👯